### PR TITLE
Sync dashboard layouts with chart configs

### PR DESCRIPTION
**Pull request by ptbrowne**
This makes sure that while we are adding and removing chart configs, when we go back to the canvas, the the charts have a correct size (previously they would be very tiny). 
Also it makes it easier to resize the last charts vertically by adding a margin bottom to the canvas layout in preview mode.

### CODE REVIEW

File: app/configurator/configurator-state/reducer.tsx [hunk 1]
```diff
@@ -490,6 +493,7 @@ export const updateColorMapping = (
       }
     }
 
+    console.log({ colorMapping, path: path.concat("colorMapping") });
```
Comment (bprusinowski) [inline app/configurator/configurator-state/reducer.tsx:496]: 🔍

File: app/configurator/configurator-state/reducer.tsx [hunk 2]
```diff
@@ -1094,13 +1102,68 @@ export const reducer_: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
 };
 
 /** Turn this on for the reducer to log state, action and result */
-const reducerLogging = false;
+const reducerLogging = true;
```
Comment (bprusinowski) [inline app/configurator/configurator-state/reducer.tsx:1105]: 🔎

File: app/configurator/configurator-state/reducer.tsx [hunk 3]
```diff
@@ -1094,13 +1102,68 @@ export const reducer_: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
...
 const withLogging = <TState, TAction extends { type: unknown }>(
   reducer: Reducer<TState, TAction>
 ) => {
   return (state: Draft<TState>, action: TAction) => {
     const res = reducer(state, action);
+    console.log(`Action: ${action.type}`, action, res);
```
Comment (bprusinowski) [inline app/configurator/configurator-state/reducer.tsx:1111]: 👁️‍🗨️

File: app/configurator/configurator-state/reducer.tsx [hunk 4]
```diff
@@ -1094,13 +1102,68 @@ export const reducer_: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
...
+  draft: WritableDraft<ConfiguratorState>
+) {
+  if (
+    hasChartConfigs(draft) &&
+    draft.layout.type === "dashboard" &&
+    draft.layout.layout === "canvas"
+  ) {
+    const chartConfigKeys = draft.chartConfigs.map((c) => c.key).sort();
+
+    const canvasLayouts = draft.layout.layouts["lg"];
```
Comment (bprusinowski) [inline app/configurator/configurator-state/reducer.tsx:1127]: Is this property always there? What if the size of the canvas was initialized as "md"?

**Review by bprusinowski — APPROVED** [review]
LGTM 👍

File: app/configurator/configurator-state/reducer.tsx (hunk 3 above)
Comment (ptbrowne) [reply app/configurator/configurator-state/reducer.tsx:1111]: This one is OK, it's when we have activated the loggin.

Comment (bprusinowski) [reply app/configurator/configurator-state/reducer.tsx:1111]: Ah, you're right 👍

File: app/configurator/configurator-state/reducer.tsx (hunk 4 above)
Comment (ptbrowne) [reply app/configurator/configurator-state/reducer.tsx:1127]: Right now this property is always there yes. The chart is always initialized with md.

Comment (ptbrowne) [reply app/configurator/configurator-state/reducer.tsx:1126]: The responsive behavior could be checked later, but I think right now it works well, I've added https://github.com/visualize-admin/visualization-tool/issues/1567

Comment (bprusinowski) [reply app/configurator/configurator-state/reducer.tsx:1126]: Thanks 💯

**Pull request merged**